cluster/src/main/java/org/apache/iotdb/cluster/query/aggregate/ClusterAggregator.java

```diff
@@ -244,12 +244,16 @@ public ClusterAggregator(MetaGroupMember metaGroupMember) {
               results);
           return results;
         }
-      } catch (TException | IOException e) {
+      } catch (TException e) {
+        logger.error(
+            metaGroupMember.getName() + " query aggregation error " + path + " from " + node, e);
+        throw new StorageEngineException(e.getMessage());
```

Comment:
any reasons ?

Reply:
In my opinion, the retrying to other nodes only happends when there're timeout exception.
If some exception, like 'UnSupportedDataTypeException' throws, there's no need to retry in other nodes, and we should tell the caller the true error message, not the outer 'TimeoutExecption'.

Reply:
> the retrying to other nodes only happends when there're timeout exception.
Agree with you.

However, I recorded that if a timeout occurs in an RPC call, TException is still thrown, and the new implementation is still not the way we want it to be. Do you have any ideas on this? Or we could refer to existing [implementations](https://github.com/apache/iotdb/blob/master/cluster/src/main/java/org/apache/iotdb/cluster/server/member/RaftMember.java#L1377), but I'm not sure whether just specifying `socketTimeoutException` is enough.

Reply:
> > the retrying to other nodes only happends when there're timeout exception.
> 
> Agree with you.
> 
> However, I recorded that if a timeout occurs in an RPC call, TException is still thrown, and the new implementation is still not the way we want it to be. Do you have any ideas on this? Or we could refer to existing [implementations](https://github.com/apache/iotdb/blob/master/cluster/src/main/java/org/apache/iotdb/cluster/server/member/RaftMember.java#L1377), but I'm not sure whether just specifying `socketTimeoutException` is enough.

Maybe we can check whether it's a TApplicationiException first, and then TException? A TApplicationException usally means there's some application error in the server side, the request-response is not timeout.

I have found another picture about the exception tree in Thrift. 
![image](https://user-images.githubusercontent.com/3821212/146164284-99c7636f-54bc-4083-8c49-b62a7ec2ea4e.png)

Reply:
I have rewrited these exception handling codes.

Reply:
good idea!